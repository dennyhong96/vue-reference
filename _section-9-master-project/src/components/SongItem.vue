<template>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition 
          duration-300 hover:bg-gray-50"
  >
    <div>
      <router-link
        :to="{ name: 'Song', params: { id: song.id } }"
        class="font-bold block text-gray-600"
        >{{ song.modifiedName }}</router-link
      >
      <span class="text-gray-500 text-sm">{{ song.displayName }}</span>
    </div>

    <div class="text-gray-600 text-lg">
      <span class="comments">
        <i class="fa fa-comments text-gray-600"></i>
        {{ song.commentCount }}
      </span>
    </div>
  </li>
</template>

<script lang="ts">
import { defineComponent, PropType } from "vue";

import { SongWithId } from "@/types/Song";

export default defineComponent({
  name: "SongItem",

  props: {
    song: {
      required: true,
      type: Object as PropType<SongWithId>,
    },
  },
});
</script>
