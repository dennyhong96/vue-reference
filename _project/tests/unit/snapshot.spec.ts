import { RouterLinkStub, shallowMount } from "@vue/test-utils";

import SongItem from "@/components/SongItem.vue";
import { TEST_SONG } from "./__mocks__";

describe("Snapshot SongItem.vue", () => {
  test("Renders correctly", () => {
    const wrapper = shallowMount(SongItem, {
      props: {
        song: TEST_SONG,
      },
      global: {
        components: {
          "router-link": RouterLinkStub,
        },
      },
    });

    // wrapper.element is the root DOM element of the component
    expect(wrapper.element).toMatchSnapshot();
  });
});
