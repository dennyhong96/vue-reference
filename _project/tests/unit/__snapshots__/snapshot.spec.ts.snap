// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot SongItem.vue Renders correctly 1`] = `
<li
  class="example-test-id flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 example snapshot"
  id="song-id-test-id"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      test-user
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom=""
      to="[object Object]"
    />
  </div>
</li>
`;
